src/components/Modal.ts

```diff
@@ -0,0 +1,140 @@
+import { RATING_MAP } from "../constants";
+import { MovieModel } from "../types/type";
+import { getUserRating, setUserRating } from "../utils/storage";
+
+export function Modal() {
+  let modalElement: HTMLElement | null = null;
+  let starsContainer: HTMLElement | null = null;
+  let ratingText: HTMLElement | null = null;
+
+  const createModalTemplate = () => {
+    const template = document.createElement("div");
+    template.innerHTML = `
+      <div class="modal-background">
+        <div class="modal">
+          <button class="close-modal">&times;</button>
+          <div class="modal-container">
+            <div class="modal-image">
+              <img src="" alt="Movie Poster">
+            </div>
+            <div class="modal-description">
+              <h2 class="movie-title"></h2>
+              <p class="category"></p>
+              <p class="rate"><strong>평점:</strong> <span class="average-rating"></span></p>
+              <div class="user-rating">
+                <p><strong>내 별점</strong></p>
+                <div class="stars"></div>
+                <p class="user-rating-text"></p>
+              </div>
+              <hr>
+              <p class="detail"></p>
+            </div>
+          </div>
+        </div>
+      </div>
+    `;
+    return template.firstElementChild as HTMLElement;
+  };
+
+  const render = () => {
+    if (!modalElement) {
+      modalElement = createModalTemplate();
+      document.body.appendChild(modalElement);
+
+      starsContainer = modalElement.querySelector(".stars");
+      ratingText = modalElement.querySelector(".user-rating-text");
+
+      modalElement
+        .querySelector(".close-modal")
+        ?.addEventListener("click", close);
+      modalElement.addEventListener("click", (e) => {
+        if (e.target === modalElement) close();
+      });
+      document.addEventListener("keydown", (e) => {
+        if (e.key === "Escape") close();
+      });
+
+      starsContainer?.addEventListener("mouseover", hoverStarRating);
+      starsContainer?.addEventListener("mouseleave", resetStarRating);
+      starsContainer?.addEventListener("click", setUserRatingClick);
+    }
+  };
```

Comment:
early return을 사용해서 depth를 줄여보셔도 좋겠네용 ㅎㅎ
```suggestion
  const render = () => {
    if (modalElement) return;

      modalElement = createModalTemplate();
      document.body.appendChild(modalElement);

      starsContainer = modalElement.querySelector(".stars");
      ratingText = modalElement.querySelector(".user-rating-text");

      modalElement
        .querySelector(".close-modal")
        ?.addEventListener("click", close);
      modalElement.addEventListener("click", (e) => {
        if (e.target === modalElement) close();
      });
      document.addEventListener("keydown", (e) => {
        if (e.key === "Escape") close();
      });

      starsContainer?.addEventListener("mouseover", hoverStarRating);
      starsContainer?.addEventListener("mouseleave", resetStarRating);
      starsContainer?.addEventListener("click", setUserRatingClick);

  };
```

Reply:
넵,,! 감사합니다